### ISSUE TITLE
Server crashes when trying to go to definition of function

### ISSUE THREAD

**Issue by jfacorro** [erlang-ls/erlang_ls#76]
**Describe the bug**
Server crashes when trying to go to definition of function.

**To Reproduce**
Open the file `jsx.erl` from the `erlang_ls` project and try to go to to the definition of `jsx_to_term:to_term/2` in line 84:
```
82: -spec decode(Source::json_text(), Config::jsx_to_term:config()) -> json_term() | {incomplete, decoder()}.
83: 
84: decode(Source, Config) -> jsx_to_term:to_term(Source, Config).
```

**Expected behavior**
Go to the definition of the function requested.

**Actual behavior**
In the logs you will see the messages:
```
2019-10-05 11:44:47.388 [error] <0.154.0>@erlang_ls_code_navigation:goto_definition:37 gen_statem <0.154.0> in state connected terminated with reason: no case clause matching [#{info => {function,{to_term,2}}
,range => #{from => {77,0},to => {77,7}},type => poi},#{info => {function,{to_term,2}},range => #{from => {73,0},to => {73,7}},type => poi}] in erlang_ls_code_navigation:goto_definition/3 line 37
2019-10-05 11:44:47.392 [error] <0.154.0>@erlang_ls_code_navigation:goto_definition:37 CRASH REPORT Process <0.154.0> with 0 neighbours crashed with reason: no case clause matching [#{info => {function,{to_te
rm,2}},range => #{from => {77,0},to => {77,7}},type => poi},#{info => {function,{to_term,2}},range => #{from => {73,0},to => {73,7}},type => poi}] in erlang_ls_code_navigation:goto_definition/3 line 37
2019-10-05 11:44:47.392 [error] <0.154.0>@erlang_ls_code_navigation:goto_definition:37 Ranch listener erlang_ls terminated with reason: no case clause matching [#{info => {function,{to_term,2}},range => #{from => {77,0},to => {77,7}},type => poi},#{info => {function,{to_term,2}},range => #{from => {73,0},to => {73,7}},type => poi}] in erlang_ls_code_navigation:goto_definition/3 line 37
```

**Context**
 - `erlang_ls` version (tag/sha): 8a6eb98fbfb570b246713de943130b042c13ddbc
 - Editor used: Emacs
 - LSP client used: lsp-mode

**Comment by jfacorro** [comment]
The root cause of this issue is the following piece of code in `jsx_to_term.erl`:
```
-ifdef(maps_always).
to_term(Source, Config) when is_list(Config) ->
    (jsx:decoder(?MODULE, [return_maps] ++ Config, jsx_config:extract_config(Config)))(Source).
-endif.
-ifndef(maps_always).
to_term(Source, Config) when is_list(Config) ->
    (jsx:decoder(?MODULE, Config, jsx_config:extract_config(Config)))(Source).
-endif.
```
Therefore there are two different "points of interest" (POIs) with a definition for the function. 🤷‍♂

**Comment by robertoaloi** [comment]
This is solved by #77. The only thing missing is a test case to avoid future regressions.
